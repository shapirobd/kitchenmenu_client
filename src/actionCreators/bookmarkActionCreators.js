import axios from "axios";
import { BOOKMARK_RECIPE, UNBOOKMARK_RECIPE } from "../components/actionTypes";

// adds bookmarked recipe to database as well as redux state
// by dispatching action created by bookmarkedRecipe()
export const bookmarkRecipe = (username, recipeId) => {
	return async (dispatch) => {
		try {
			await axios.post(
				`${
					process.env.REACT_APP_API_URL || "http://localhost:5000"
				}/users/bookmarkRecipe`,
				{
					username,
					recipeId,
				}
			);
			dispatch(bookmarkedRecipe(recipeId));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing id of recipe to be bookmarked
const bookmarkedRecipe = (recipeId) => {
	return {
		type: BOOKMARK_RECIPE,
		payload: {
			recipeId,
		},
	};
};

// removes bookmarked recipe from database as well as redux state
// by dispatching action created by unbookmarkedRecipe()
export const unbookmarkRecipe = (username, recipeId) => {
	return async (dispatch) => {
		try {
			await axios.post(
				`${
					process.env.REACT_APP_API_URL || "http://localhost:5000"
				}/users/unbookmarkRecipe`,
				{
					username,
					recipeId,
				}
			);
			dispatch(unbookmarkedRecipe(recipeId));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing id of recipe to be unbookmarked
const unbookmarkedRecipe = (recipeId) => {
	return {
		type: UNBOOKMARK_RECIPE,
		payload: {
			recipeId,
		},
	};
};
