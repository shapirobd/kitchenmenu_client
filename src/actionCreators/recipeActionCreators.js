import axios from "axios";
import { LOAD_FEED, LOAD_RECIPE, FILTER_FEED } from "../components/actionTypes";
import createMacrosParams from "../helpers/createMacrosParams";

// retrieves all recipes from spoonacular API (paginated 40 per page) and updates
// the feed & page number in redux state by dispatching array of recipes and the page number
export const loadFeed = (page = 1, filterData = { diets: [], macros: {} }) => {
	return async (dispatch) => {
		try {
			const { diets, macros, recipeName } = filterData;
			const macrosParams = createMacrosParams(macros);
			const dietsParams = diets.join(",");
			const recipes = await axios.get(
				"https://api.spoonacular.com/recipes/complexSearch",
				{
					params: {
						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
						offset: 40 * (page - 1),
						number: 40,
						diet: dietsParams,
						...macrosParams,
						query: recipeName,
					},
				}
			);
			dispatch(loadedFeed(recipes.data, page));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action containing recipes to include in redux state's feed,
// total number of results found to set as redux state's totalResults,
// and the current page number to set as redux state's page
const loadedFeed = (data, page) => {
	return {
		type: LOAD_FEED,
		payload: {
			recipes: data.results,
			totalResults: data.totalResults <= 900 ? data.totalResults : 900,
			page,
		},
	};
};

// retrieves data on a single recipe from spoonacular API and updates
// currentRecipe in redux state by dispatching recipe info & instructions info
export const loadRecipe = (recipeId) => {
	console.log("2: ", recipeId);
	return async (dispatch) => {
		try {
			const recipe = await axios.get(
				`https://api.spoonacular.com/recipes/${recipeId}/information`,
				{
					params: {
						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
						includeNutrition: true,
					},
				}
			);
			const instructions = await axios.get(
				`https://api.spoonacular.com/recipes/${recipeId}/analyzedInstructions`,
				{
					params: {
						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
						stepBreakdown: true,
					},
				}
			);
			dispatch(loadedRecipe(recipe.data, instructions.data[0]));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action containing information to include in redux state's currentRecipe
const loadedRecipe = (recipe, instructions) => {
	return {
		type: LOAD_RECIPE,
		payload: { currentRecipe: { recipe, instructions } },
	};
};

// retrieves all recipes from spoonacular API with given ingredients (pagination not an option)
// and updates feed, totalResults, page, and isIngredientBased in redux state by
// dispatching array of found recipes and the length of that array
export const getRecipesByIngredients = (ingredients) => {
	return async (dispatch) => {
		try {
			const ingredientsParams = ingredients.join(",");
			const recipes = await axios.get(
				"https://api.spoonacular.com/recipes/findByIngredients",
				{
					params: {
						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
						offset: 0,
						number: 6900,
						ingredients: ingredientsParams,
					},
				}
			);
			dispatch(
				filteredFeed({ results: recipes.data, totalResults: recipes.length })
			);
		} catch (e) {
			console.error(e);
		}
	};
};

// export const getRecipesByKeywords = (keywords) => {
// 	return async (dispatch) => {
// 		try {
// 			const ingredientsParams = ingredients.join(",");
// 			const recipes = await axios.get(
// 				"https://api.spoonacular.com/recipes/findByIngredients",
// 				{
// 					params: {
// 						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
// 						offset: 0,
// 						number: 6900,
// 						query: keywords,
// 					},
// 				}
// 			);
// 			dispatch(
// 				filteredFeed({ results: recipes.data, totalResults: recipes.length })
// 			);
// 		} catch (e) {
// 			console.error(e);
// 		}
// 	};
// };

// const gotRecipesByIngredients = (results) => {
// 	return {
// 		type: FILTER_BY_INGREDIENTS,
// 		payload: { recipes: data.results, totalResults: data.totalResults },
// 	};
// };

// export const filterFeed = (filterData, page = 1) => {
// 	return async (dispatch) => {
// 		try {
// 			const { diets, macros, recipeName } = filterData;
// 			const macrosParams = createMacrosParams(macros);
// 			const dietsParams = diets.join(",");
// 			const recipes = await axios.get(
// 				"https://api.spoonacular.com/recipes/complexSearch",
// 				{
// 					params: {
// 						apiKey: "73baf9bb95a14f5fb4d71e2f12ab8479",
// 						offset: 40 * (page - 1),
// 						number: 40,
// 						diet: dietsParams,
// 						...macrosParams,
// 						query: recipeName,
// 					},
// 				}
// 			);
// 			console.log(recipes.data);
// 			dispatch(filteredFeed(recipes.data, page));
// 		} catch (e) {
// 			console.error(e);
// 		}
// 	};
// };

// returns action containing information to include in redux state's feed, totalResults, and page
const filteredFeed = (data, page) => {
	return {
		type: FILTER_FEED,
		payload: { recipes: data.results, totalResults: data.totalResults, page },
	};
};
