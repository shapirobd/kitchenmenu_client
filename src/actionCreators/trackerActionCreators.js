import axios from "axios";
import { ADD_EATEN_MEAL, REMOVE_EATEN_MEAL } from "../components/actionTypes";
import { API_URL } from "../constants";

// const API_URL = "http://localhost:5000";

// adds eaten meal to database as well as redux state for a given date
// by dispatching action created by addedEatenMeal()
export const addEatenMeal = (username, recipeId, date, nutrients) => {
	return async (dispatch) => {
		try {
			const macros = {
				calories: Math.round(nutrients[0].amount),
				fat: Math.round(nutrients[1].amount),
				carbs: Math.round(nutrients[3].amount),
				protein: Math.round(nutrients[8].amount),
			};
			await axios.post(`${API_URL}/users/addEatenMeal`, {
				username,
				recipeId,
				date,
				nutrients: macros,
			});
			dispatch(addedEatenMeal(recipeId, date, macros));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing id of recipe
// to be eaten and date that it was eaten
const addedEatenMeal = (recipeId, date, nutrients) => {
	const { calories, fat, carbs, protein } = nutrients;
	return {
		type: ADD_EATEN_MEAL,
		payload: {
			date,
			meal: {
				id: recipeId,
				calories,
				fat,
				carbs,
				protein,
			},
		},
	};
};

// removes eaten meal from database as well as redux state for a given date
// by dispatching action created by removedEatenMeal()
export const removeEatenMeal = (username, recipeId, date) => {
	return async (dispatch) => {
		try {
			await axios.post(`${API_URL}/users/removeEatenMeal`, {
				username,
				recipeId,
				date,
			});
			dispatch(removedEatenMeal(recipeId, date));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing id of recipe
// to be removed and date that it was eaten
const removedEatenMeal = (recipeId, date) => {
	return {
		type: REMOVE_EATEN_MEAL,
		payload: {
			recipeId,
			date,
		},
	};
};
