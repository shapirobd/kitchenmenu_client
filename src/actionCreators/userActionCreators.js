import axios from "axios";
import { LOGIN, LOGOUT } from "../components/actionTypes";

// adds user to database as well as redux state
// by dispatching action created by registered()
export const register = (data) => {
	return async (dispatch) => {
		try {
			const resp = await axios.post(
				`${
					process.env.REACT_APP_API_URL || "http://localhost:5000"
				}/auth/register`,
				data
			);
			// const userResp = await axios.get(`${API_URL}/users/${data.username}`);
			const { user, token } = resp.data;
			dispatch(registered(token, user));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing information on the
// newly registered user
const registered = (token, user) => {
	return {
		type: LOGIN,
		payload: {
			user,
			token,
		},
	};
};

// verifies that username & password from data are valid
// if valid, adds the user to redux state by dispatching action created by loggedIn()
// in invalid, throws error
export const login = (data) => {
	return async (dispatch) => {
		try {
			const resp = await axios.post(
				`${
					process.env.REACT_APP_API_URL || "http://localhost:5000"
				}/auth/login`,
				data
			);
			const { user, token } = resp.data;
			dispatch(loggedIn(token, user));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing information on the
// logged in user
const loggedIn = (token, user) => {
	return {
		type: LOGIN,
		payload: {
			user,
			token,
		},
	};
};

// returns action to be dispatched that ends up removing
// the user's information from redux state
export const logout = () => {
	return {
		type: LOGOUT,
	};
};

// edits the information on a given user in the database
// as well as redux state by dispatching action created by edittedProfile()
export const editProfile = (username, data) => {
	return async (dispatch) => {
		try {
			const editResp = await axios.patch(
				`${process.env.API_URL || "http://localhost:5000"}/users/${username}`,
				data
			);
			dispatch(edittedProfile(editResp.data));
		} catch (e) {
			console.error(e);
		}
	};
};

// returns action to be dispatched containing information on the
// logged in user
const edittedProfile = (user) => {
	return {
		type: "EDIT_PROFILE",
		payload: { user },
	};
};
