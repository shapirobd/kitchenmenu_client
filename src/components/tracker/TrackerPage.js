import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import { Grid } from "@material-ui/core";
import { useStyles } from "./styles/TrackerPageStyles";
import "react-calendar/dist/Calendar.css";
import convertDate from "../../helpers/convertDate";
import { getPieChartData, getDateMacros } from "./helpers/dailyHelpers";
import {
	getBarChartData,
	getWeekNumber,
	getWeekDates,
	getWeekMacros,
} from "./helpers/weeklyHelpers";
import TrackerCalendar from "./TrackerCalendar";
import TrackerDoughnut from "./TrackerDoughnut";
import TrackerBarChart from "./TrackerBarChart";
import useWindowDimensions from "../../customHooks/getWindowDimensions";

const TrackerPage = () => {
	const classes = useStyles();
	const { width } = useWindowDimensions();

	const user = useSelector((state) => state.user);

	const [calendarDate, setCalendarDate] = useState(new Date());
	const [dayState, setDayState] = useState({
		loaded: false,
		empty: false,
		date: convertDate(calendarDate),
		macros: {
			carbs: null,
			fat: null,
			protein: null,
		},
	});
	const [weekState, setWeekState] = useState({
		loaded: false,
		empty: false,
		weekNumber: null,
		dates: getWeekDates(calendarDate),
		weekData: {},
	});
	const [pieChartData, setPieChartData] = useState(undefined);
	const [barChartData, setBarChartData] = useState(undefined);

	useEffect(() => {
		setPieChartData(getPieChartData());
	}, []);

	useEffect(() => {
		setBarChartData(getBarChartData(weekState.weekData));
	}, [weekState.weekData]);

	useEffect(() => {
		const updateWeekState = async () => {
			getDateMacros(user, "day", convertDate(calendarDate), setDayState);
			const weekData = await getWeekMacros(user, getWeekDates(calendarDate));
			let empty = true;
			Object.values(weekData).map((date) => {
				if (empty && (date.carbs || date.fat || date.protein)) {
					empty = false;
				}
			});
			if (getWeekNumber(calendarDate) !== weekState.weekNumber) {
				setWeekState({
					empty,
					loaded: true,
					weekNumber: getWeekNumber(calendarDate),
					dates: getWeekDates(calendarDate),
					weekData,
				});
			}
		};
		updateWeekState();
	}, [calendarDate, weekState, user]);

	useEffect(() => {
		setPieChartData(
			getPieChartData(
				dayState.macros.carbs,
				dayState.macros.fat,
				dayState.macros.protein
			)
		);
	}, [dayState]);

	useEffect(() => {
		setBarChartData(getBarChartData(weekState.weekData));
	}, [weekState.weekData]);

	return (
		<div style={{ width: "100%" }}>
			<Grid
				container
				cols={width <= 599 ? 1 : 2}
				className={width <= 599 ? classes.mobileRoot : classes.root}
			>
				<Grid item cols={1} md={4} className={classes.gridItem}>
					<TrackerCalendar
						calendarDate={calendarDate}
						setCalendarDate={setCalendarDate}
					/>

					{width > 599 ? (
						<TrackerDoughnut dayState={dayState} pieChartData={pieChartData} />
					) : null}
				</Grid>
				{width <= 599 ? (
					<Grid item cols={1} md={4} className={classes.gridItem}>
						<TrackerDoughnut dayState={dayState} pieChartData={pieChartData} />
					</Grid>
				) : null}
				<Grid item cols={1} md={8} className={classes.gridItem}>
					<TrackerBarChart weekState={weekState} barChartData={barChartData} />
				</Grid>
			</Grid>
		</div>
	);
};

export default TrackerPage;
