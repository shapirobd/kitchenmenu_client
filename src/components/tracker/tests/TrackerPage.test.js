import React from "react";
import { render, screen, act } from "@testing-library/react";
import TrackerPage from "../TrackerPage";
import { MemoryRouter } from "react-router-dom";
import { PersistGate } from "redux-persist/integration/react";
import { Provider } from "react-redux";
import { store, persistor } from "../../../configureStore";
import { login } from "../../../actionCreators/userActionCreators";

beforeAll(async () => {
	await store.dispatch(
		login({ username: "shapirobd", password: "Pot3ntiat321!" })
	);
});

it("should render without crashing", () => {
	render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<TrackerPage />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);
});

it("should match screenshot", () => {
	const { asFragment } = render(
		<Provider store={store}>
			<MemoryRouter>
				<PersistGate loading={null} persistor={persistor}>
					<TrackerPage />
				</PersistGate>
			</MemoryRouter>
		</Provider>
	);

	expect(asFragment()).toMatchSnapshot();
});
