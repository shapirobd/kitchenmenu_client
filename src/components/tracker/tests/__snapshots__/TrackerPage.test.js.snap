// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match screenshot 1`] = `
<DocumentFragment>
  <div
    style="width: 100%;"
  >
    <div
      class="MuiGrid-root makeStyles-root-15 MuiGrid-container"
      cols="2"
    >
      <div
        class="MuiGrid-root makeStyles-gridItem-17 MuiGrid-item MuiGrid-grid-md-4"
        cols="1"
      >
        <div
          class="makeStyles-calendarDiv-21"
        >
          <div
            class="react-calendar makeStyles-calendar-20"
          >
            <div
              class="react-calendar__navigation"
              style="display: flex;"
            >
              <button
                aria-label=""
                class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
                type="button"
              >
                «
              </button>
              <button
                aria-label=""
                class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
                type="button"
              >
                ‹
              </button>
              <button
                aria-label=""
                class="react-calendar__navigation__label"
                style="flex-grow: 1;"
                type="button"
              >
                <span
                  class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
                >
                  March 2021
                </span>
              </button>
              <button
                aria-label=""
                class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
                type="button"
              >
                ›
              </button>
              <button
                aria-label=""
                class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
                type="button"
              >
                »
              </button>
            </div>
            <div
              class="react-calendar__viewContainer"
            >
              <div
                class="react-calendar__month-view "
              >
                <div
                  style="display: flex; align-items: flex-end;"
                >
                  <div
                    style="flex-grow: 1; width: 100%;"
                  >
                    <div
                      class="react-calendar__month-view__weekdays"
                      style="display: flex; flex-wrap: no-wrap;"
                    >
                      <div
                        class="react-calendar__month-view__weekdays__weekday"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      >
                        <abbr
                          aria-label="Sunday"
                          title="Sunday"
                        >
                          Sun
                        </abbr>
                      </div>
                      <div
                        class="react-calendar__month-view__weekdays__weekday"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      >
                        <abbr
                          aria-label="Monday"
                          title="Monday"
                        >
                          Mon
                        </abbr>
                      </div>
                      <div
                        class="react-calendar__month-view__weekdays__weekday"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      >
                        <abbr
                          aria-label="Tuesday"
                          title="Tuesday"
                        >
                          Tue
                        </abbr>
                      </div>
                      <div
                        class="react-calendar__month-view__weekdays__weekday"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      >
                        <abbr
                          aria-label="Wednesday"
                          title="Wednesday"
                        >
                          Wed
                        </abbr>
                      </div>
                      <div
                        class="react-calendar__month-view__weekdays__weekday"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      >
                        <abbr
                          aria-label="Thursday"
                          title="Thursday"
                        >
                          Thu
                        </abbr>
                      </div>
                      <div
                        class="react-calendar__month-view__weekdays__weekday"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      >
                        <abbr
                          aria-label="Friday"
                          title="Friday"
                        >
                          Fri
                        </abbr>
                      </div>
                      <div
                        class="react-calendar__month-view__weekdays__weekday"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      >
                        <abbr
                          aria-label="Saturday"
                          title="Saturday"
                        >
                          Sat
                        </abbr>
                      </div>
                    </div>
                    <div
                      class="react-calendar__month-view__days"
                      style="display: flex; flex-wrap: wrap;"
                    >
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="February 28, 2021"
                        >
                          28
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 1, 2021"
                        >
                          1
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 2, 2021"
                        >
                          2
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 3, 2021"
                        >
                          3
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 4, 2021"
                        >
                          4
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 5, 2021"
                        >
                          5
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 6, 2021"
                        >
                          6
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 7, 2021"
                        >
                          7
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 8, 2021"
                        >
                          8
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 9, 2021"
                        >
                          9
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 10, 2021"
                        >
                          10
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 11, 2021"
                        >
                          11
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 12, 2021"
                        >
                          12
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 13, 2021"
                        >
                          13
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 14, 2021"
                        >
                          14
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 15, 2021"
                        >
                          15
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 16, 2021"
                        >
                          16
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 17, 2021"
                        >
                          17
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 18, 2021"
                        >
                          18
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 19, 2021"
                        >
                          19
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 20, 2021"
                        >
                          20
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 21, 2021"
                        >
                          21
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 22, 2021"
                        >
                          22
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 23, 2021"
                        >
                          23
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 24, 2021"
                        >
                          24
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 25, 2021"
                        >
                          25
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 26, 2021"
                        >
                          26
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__tile--now react-calendar__tile--active react-calendar__tile--range react-calendar__tile--rangeStart react-calendar__tile--rangeEnd react-calendar__tile--rangeBothEnds react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 27, 2021"
                        >
                          27
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 28, 2021"
                        >
                          28
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 29, 2021"
                        >
                          29
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 30, 2021"
                        >
                          30
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="March 31, 2021"
                        >
                          31
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="April 1, 2021"
                        >
                          1
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="April 2, 2021"
                        >
                          2
                        </abbr>
                      </button>
                      <button
                        class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                        style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                        type="button"
                      >
                        <abbr
                          aria-label="April 3, 2021"
                        >
                          3
                        </abbr>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-doughnutDiv-24"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Daily Macros
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorError"
          >
            No data for the selected date
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-gridItem-17 MuiGrid-item MuiGrid-grid-md-8"
        cols="1"
      >
        <div
          class="makeStyles-barChartDiv-27"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
